### DISCUSSION TITLE
Sofar KTLM-G3 inverter - newbie help?

### DISCUSSION THREAD

**Discussion by stephencrowther** [davidrapan/ha-solarman#230] [Q&A] [answered]
Can anyone give me some simple to follow instructions on how to pull data from this basic sofar inverter into HA.

I tried using the MY link on this git and downloading this into HACS.

Got the form for serial number and IP address (which got pre filled with the correct info somehow).

But then after that I get an error and non working integration.

**Comment by davidrapan** [comment]
Hi @stephencrowther, maybe #172 will help?

**Reply by stephencrowther** [reply]
Have read through all that - quite complicated to follow other than it seems to suggest getting the firmware updated on the inverter?

Also not strictly the same model of inverter.

**Reply by davidrapan** [reply]
I'm pretty sure that your debug log will also contain the same CRC Error. 😉

**Reply by stephencrowther** [reply]
My problem doesn't seem to match the problem in the thread you referenced though. It's like it's failing to set up, rather than setting up and not being able to pull any data from the inverter. I can't see any 'CRC error' anywhere in the log.
<img width="1024" alt="Screenshot 2024-10-21 at 00 31 20" src="https://github.com/user-attachments/assets/0a67ff54-aaa3-4a81-9dc2-6103bc80f297">
<img width="1397" alt="Screenshot 2024-10-21 at 00 40 39" src="https://github.com/user-attachments/assets/881dab1a-e898-4686-98d8-414bd282a80d">

**Reply by davidrapan** [reply, marked as answer]
Then it's this one #39.

ssh into your HA and run `ha core rebuild`.

**Reply by stephencrowther** [reply]
ok that's solved that bit - thanks for pointing me to that!

now I get the CRC error - inverter is pulling values but looks like its pulling the wrong values for each entity, so now I guess its get the inverter firmware updated which seems to be the key action in 172.

**Reply by davidrapan** [reply]
Yes I would say so.
